src/api_types.jl

```diff
@@ -0,0 +1,307 @@
+## C types
+const C_time_t = Int
+
+## HDF5 types and constants
+const haddr_t  = UInt64
+const herr_t   = Cint
+const hid_t    = Int64
+const hsize_t  = UInt64
+const hssize_t = Int64
+const htri_t   = Cint   # pseudo-boolean (negative if error)
+
+# For VLEN
+struct hvl_t
+    len::Csize_t
+    p::Ptr{Cvoid}
+end
+const HVL_SIZE = sizeof(hvl_t)
+
+# Object reference types
+struct HDF5ReferenceObj
+    r::UInt64 # Size must be H5R_OBJ_REF_BUF_SIZE
+end
+const HDF5ReferenceObj_NULL = HDF5ReferenceObj(UInt64(0))
```

Comment:
Unrelated by I think we should just get rid of this and and use 
```
HDF5ReferenceObj(0)
```
on line 
https://github.com/JuliaIO/HDF5.jl/blob/master/src/HDF5.jl#L1521

Reply:
nvm just saw your other PR lol